spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1594,6 +1598,34 @@ public void onMessage(Message message) {
 		}
 	}
 
+	private QueueingConsumer createQueueingConsumer(final String queueName, Channel channel) throws Exception {
+		channel.basicQos(1);
+		final CountDownLatch latch = new CountDownLatch(1);
+		QueueingConsumer consumer = new QueueingConsumer(channel) {
+
+			@Override
+			public void handleCancel(String consumerTag) throws IOException {
+				super.handleCancel(consumerTag);
+				latch.countDown();
+			}
+
+			@Override
+			public void handleConsumeOk(String consumerTag) {
+				super.handleConsumeOk(consumerTag);
+				latch.countDown();
+			}
+
+		};
+		channel.basicConsume(queueName, consumer);
+		if (!latch.await(10, TimeUnit.SECONDS)) {
```

Comment:
@garyrussell , I wonder if we can get rid of this blocker.
For example `gradlew clean testAll --parallel` passes for me with this `if`.

Having this the code
```
if (receiveTimeout < 0) {
    delivery = consumer.nextDelivery();
}
```
does not make sense.
Because we end up with the `AmqpException("Blocking receive, consumer failed to consume")` exactly after this `10 sec`.

Have I missed anything?